pkg/apiserver/authentication/token/issuer.go

```diff
@@ -16,16 +16,23 @@ limitations under the License.
 
 package token
 
-import "time"
+import (
+	"k8s.io/apiserver/pkg/authentication/user"
+	"time"
+)
+
+const (
+	AccessToken  TokenType = "access_token"
+	RefreshToken TokenType = "refresh_token"
+)
+
+type TokenType string
 
 // Issuer issues token to user, tokens are required to perform mutating requests to resources
 type Issuer interface {
 	// IssueTo issues a token a User, return error if issuing process failed
-	IssueTo(user User, expiresIn time.Duration) (string, error)
-
-	// Verify verifies a token, and return a User if it's a valid token, otherwise return error
-	Verify(string) (User, error)
+	IssueTo(user user.Info, tokenType TokenType, expiresIn time.Duration) (string, error)
 
-	// Revoke a token,
-	Revoke(token string) error
+	// Verify verifies a token, and return a user info if it's a valid token, otherwise return error
+	Verify(string) (user.Info, *Claims, error)
```

Comment:
Why verify need to return `Claims` ？

Reply:
The token type needs to be returned after token validation, I made some changes.